### PR TITLE
More Tests

### PR DESCRIPTION
**Pull request by suv27**
### Functions  Tested
- [x] is_renamed_exercise()
- [x] is_shortname_exercise()
- [x] type_to_presentation_type()
- [x] alert_ignored_files()
- [x] alert_renamings_to_be_applied()

I will work on the next tests later on. This PR is ready.
Coverage Report for `vandor_rename.py` increased to 28%.
image

### CODE REVIEW

**Comment by icaropires** [comment]
Thanks for your efforts @Physsix27 ! But I'll point some things.

File: tests/test_vandor_rename.py [hunk 1]
```diff
@@ -34,19 +35,22 @@ def test_valid_names_and_exts():
 
 def test_is_renamed_exercise():
     """
-    Should ensure the type return by is_renamed_exercise() is a bool
+    Should ensure the type returned by is_renamed_exercise() is a bool
```
Comment (icaropires) [inline tests/test_vandor_rename.py]: We have types annotations on functions and there exists tools which check if types are correct automatically. So I don't think we need any tests for checking types :sweat_smile: .

We need tests only to check if exceptions are raised given inputs and if output is correct given for different inputs.

File: tests/test_vandor_rename.py [hunk 2]
```diff
@@ -34,19 +35,22 @@ def test_valid_names_and_exts():
...
+            type(vandor.is_renamed_exercise(filee))
         )
+        
 
 
-def test_return_value_of_shortname_excersise():
+def test_is_shortname_exercise():
     """
-    Should ensure the type return by is_renamed_exercise() is a bool
+    Should ensure the type returned by is_shortname_exercise() is a bool
```
Comment (icaropires) [inline tests/test_vandor_rename.py]: We have types annotations on functions and there exists tools which check if types are correct automatically. So I don't think we need any tests for checking types :sweat_smile: .

We need tests only to check if exceptions are raised given inputs and if output is correct given for different inputs.

File: tests/test_vandor_rename.py [hunk 3]
```diff
@@ -56,17 +60,56 @@ def test_return_value_of_shortname_excersise():
...
-def test_type_to_presentation_type():
-    pass
+def test_type_to_presentation_type_return1():
+    """
+    Should ensure the type and value returned by is_shortname_exercise() is a Test & str
+    """
+
+    result = vandor.type_to_presentation_type('test')
+    expected_result = 'Test'
+    assert result == expected_result
```
Comment (icaropires) [inline tests/test_vandor_rename.py]: Make this and `test_type_to_presentation_type_return2` a unique function. You can use `parametrize test` for this, look [here](https://docs.pytest.org/en/latest/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions) for instructions.

File: tests/test_vandor_rename.py [hunk 4]
```diff
@@ -56,17 +60,56 @@ def test_return_value_of_shortname_excersise():
...
 
 
 def test_alert_ignored_files():
-    pass
+    """
+    Ensures the functions prints, nothing to test in this functions
+    """
+    
+    assert vandor.alert_ignored_files(("A", 2, "C", 4)) == None
```
Comment (icaropires) [inline tests/test_vandor_rename.py]: This doesn't check if something is printed, `alert_ignored_files()` will **always** return None.

File: tests/test_vandor_rename.py [hunk 5]
```diff
@@ -56,17 +60,56 @@ def test_return_value_of_shortname_excersise():
...
+    """
+    Ensures alert_renamings_to_be_appliedalert_renamings_to_be_applied() prints
+    """
+    file_list = {
+        "oldName": "newName",
+        "Star": "*",
+        "test": "test"
+    }
+    
+    assert vandor.alert_renamings_to_be_applied(file_list) == None
```
Comment (icaropires) [inline tests/test_vandor_rename.py]: This doesn't check if something is printed, `test_alert_renamings_to_be_applied()` will always return None.

File: tests/test_vandor_rename.py [hunk 6]
```diff
@@ -56,17 +60,56 @@ def test_return_value_of_shortname_excersise():
...
+
 
+def test_no_alert_renamings_to_be_applied():
+    """
+    Ensures alert_renamings_to_be_applied() raises a FileNotFoundError
+    """
+    
+    with pytest.raises(FileNotFoundError):
```
Comment (icaropires) [inline tests/test_vandor_rename.py:113]: This test is good!

**Review by icaropires — CHANGES_REQUESTED** [review]
Thanks for the PR ! Alert me once you've finished the fixings :wink: . Also, I don't think that open another PR just for finishing issue #11 is a good idea :confused:

File: tests/test_vandor_rename.py (hunk 1 above)
Comment (suv27) [reply tests/test_vandor_rename.py]: You don't have exceptions in all of your functions if there are no exceptions I will make tests for their return values

**Comment by suv27** [comment]
I need some help on testing `print() I am not sure how can I do that. How do I test that `print` has been call?

**Comment by icaropires** [comment]
> I need some help on testing `print() I am not sure how can I do that. How do I test that `print` has been call?

@Physsix27 you can use `capfd` or `capsys` fixtures, more information [here](https://docs.pytest.org/en/latest/capture.html)

**Comment by suv27** [comment]
I am trying to use that but the thing is the way that the print is format it is killing me. I am referring to `test_alert_renamings_to_be_applied()` which is the one I am trying first.
image

**Comment by icaropires** [comment]
> I am trying to use that but the thing is the way that the print is format it is killing me.
> image

Backslashes rarely are a cool thing in Python. You can use multi-line strings like:
``` python
"""This
is
a multiline string
"""
```

Or like is done at the source code:
``` python
    print('======================================\n'
          '\tRenamings to be applied:\n'
          '======================================')
```

For getting the exact match, I recommend you look into the source code to see the details on how the strings are assembled.

**Comment by Physsix27** [comment]
Can you declare a variable called `result_exected = ` and write to me that multi-line. I can't seem to solve that issue.

**Comment by suv27** [comment]
Fixed the tests for those print statements. Its anything else needs it for this PR to be fixed. Cant work on the other tests right at this moment.

File: vandor_rename.py [hunk 7]
```diff
@@ -52,7 +52,7 @@ def alert_ignored_files(ignored_files: List[str]) -> None:
     ignored_files = '\n'.join(
         f'  {i+1}. {f}' for i, f in enumerate(ignored_files)
     )
-    print("The following files will be ignored because they don't"
```
Comment (icaropires) [inline vandor_rename.py]: I believe the correct form is `don't`

File: tests/test_vandor_rename.py [hunk 8]
```diff
@@ -32,40 +32,90 @@ def test_valid_names_and_exts():
...
+    result_expected = '''=============================
+\tIgnored Files
+=============================
+The following files will be ignored because they dont look like exercises:
+  1. A
+  2. 2
+  3. C
+  4. 4\n'''
+
+    vandor.alert_ignored_files(("A", 2, "C", 4))
```
Comment (icaropires) [inline tests/test_vandor_rename.py]: Add as input some files that should not be ignored mixed with those you already added.

File: tests/test_vandor_rename.py [hunk 9]
```diff
@@ -32,40 +32,90 @@ def test_valid_names_and_exts():
...
-        assert not isinstance(
-            type(True),
-            type(vandor.is_shortname_exercise(filee))
-        )
+    result = vandor.is_renamed_exercise(input)
+    assert expected_result == result
 
 
-def test_return_value_of_shortname_excersise():
+@pytest.mark.parametrize("input, expected_result", [("logico.brM3", True), ("fisico.sql", True)])
```
Comment (icaropires) [inline tests/test_vandor_rename.py]: Add some test case that the function should return `False`

File: tests/test_vandor_rename.py [hunk 10]
```diff
@@ -32,40 +32,90 @@ def test_valid_names_and_exts():
         assert extension == extensions
 
 
-def test_is_renamed_exercise():
+@pytest.mark.parametrize("input, expected_result", [("aula_test.brM3", True), ("fisico.sql", False)])
```
Comment (icaropires) [inline tests/test_vandor_rename.py]: Add some test case that is a real name for a renamed_exercise, like: `aula1exer2Evolucao3Consulta_AlunoSobrenome_15-0129815.sql`

**Review by icaropires — CHANGES_REQUESTED** [review]
Great improvement!

Can add more these changes? The main point was missing test cases for different situations.
Addionally, I'd like to ask for you to rename `input` to just `in`, as input is a reserved keyword in python.

I believe that with this we are good to go. Ping me once you've finished :wink:

File: tests/test_vandor_rename.py [hunk 11]
```diff
@@ -32,40 +32,90 @@ def test_valid_names_and_exts():
...
-def test_type_to_presentation_type():
-    pass
+@pytest.mark.parametrize("input, expected_result", [("test", "Test"), ("doc", "DOC")])
+def test_type_to_presentation_type(input, expected_result):
+    """
+    Should ensure the type and return value of is_shortname_exercise()
+    """
 
+    result = vandor.type_to_presentation_type(input)
+    assert expected_result, result
```
Comment (icaropires) [inline tests/test_vandor_rename.py]: You must replace `,` with `==`

**Comment by suv27** [comment]
`in` is a special keyword in python not `input`

**Comment by icaropires** [comment]
> `in` is a special keyword in python not `input`

@Physsix27 you're right. `in` is also reserved, but `input` is the function `input()` :sweat_smile: . You should use the name of the parameter, for example `renamings` for `alert_renamings_to_be_applied` function. But `data` is ok for now.

File: tests/test_vandor_rename.py (hunk 8 above)
Comment (icaropires) [reply tests/test_vandor_rename.py]: No, you were right, only invalid files should go here. But I can fix this.

**Comment by icaropires** [comment]
Thanks for contributing!

**Pull request merged**